hls4ml/writer/vivado_writer.py

```diff
@@ -297,7 +297,9 @@ def write_defines(self, model):
             if '//hls-fpga-machine-learning insert numbers' in line:
                 newline = line
                 numbers = OrderedDict.fromkeys([layer.get_numbers_cpp() for layer in model.get_layers()])
-                newline += ''.join(numbers)
+                numbers = set('\n'.join(numbers).split('\n')) #we want a unique set of macro declarations, since some of the macros are shared between different NN-blocks
+                newline += '\n'.join([i for i in numbers if i!=''])
+                newline += '\n' #for formatting purposes
```

Comment:
What's this for?

Reply:
Some macros, such as N_EDGE or N_NODE, are shared between different hls4ml layers. This basically keeps those macros from being defined several times in the cpp header.